modules/ROOT/pages/customer-connector-upgrade.adoc

```diff
@@ -0,0 +1,162 @@
+= Customer Guide for Upgrading and Testing Custom Connectors for Java 17
+
+Upgrade and test your custom connectors for Java 17 to ensure compatibility within the MuleSoft ecosystem. 
+
+== Before You Begin
+
+Before upgrading and testing your custom connector for Java 17, you must be familiar with the following:
+
+* *How to run and compile your custom connector*
++
+Ensure your custom connector runs with Java 17, but always use Java 8 as the target compilation level.
++
+Because you must compile your custom connector with Java 8, use the Java compiler configuration from the Java SDK parent POM. You can’t use language features from Java 11 or Java 17. 
+
+* *-add-opens or -add-export directives*
++
+Do not set `-add-opens` or `-add-export` directives.
+
+* *Minimum Mule version caveats*
++
+If your custom connector is running on Mule 4.5.x or earlier, you can’t run Java 17 tests that extend the MuleArtifactFunctionalTestCase, but you can still run Java 8 tests.
++
+To decouple the minimum Mule version when running tests, add the `-Dmunit.jvm` parameter to use Java 17 as described in <<test-your-custom-connector-with-munit>>.
+
+[[upgrade-your-mule-sdk-connectors]]
+== Upgrade Your Custom Mule SDK Connectors
+
+Upgrade your custom Mule SDK (Java SDK and XML SDK) connectors to Java 17.
+
+=== Upgrade Third-Party Libraries
+
+Upgrade your custom connector’s third-party libraries to be compatible with Java 17.
+
+* Java EE libraries
++
+Mule 4.6 exposes Java EE libraries differently. You must add the following BOM dependency to your custom connector and, if applicable, ensure the provided conflicting library is properly excluded. You can also add the libraries in the BOM dependency separately.
```

Comment:
differently than...what?

Reply:
"exclude the provided conflicting library."?

Reply:
I sort of want an intro sentence. Remove "following" from second sentence and add an intro?
Add the following BOM ... :

Reply:
previous: Mule 4.6 exposes Java EE libraries differently. You must add the following BOM dependency to your custom connector and, if applicable, ensure the provided conflicting library is properly excluded. You can also add the libraries in the BOM dependency separately.

now: Mule 4.6 exposes Java EE libraries differently than earlier Mule versions. You must add the BOM dependency to your custom connector and, if applicable, exclude the provided conflicting library. You can also add the libraries in the BOM dependency separately.

Reply:
That's great!